data/data.go

```diff
@@ -29,6 +31,23 @@ type EffxYaml struct {
 	FilePath string
 }
 
+func setMetadata(config *effx_api.ConfigurationFile, m *metadata.Result) *effx_api.ConfigurationFile {
+	if m != nil {
+		if config.Annotations == nil {
+			config.Annotations = &map[string]string{}
+		}
+		if config.Tags == nil {
+			config.Tags = &map[string]string{}
+		}
+
+		(*config.Annotations)["effx.io/inferred-tags"] = fmt.Sprintf("language,%s", m.Language)
+		(*config.Tags)["language"] = m.Language
```

Comment:
I'm concerned if the user has already defined a language field. we should take theirs instead. can we add some logic to only add the annotation and tags if they are undefined?

Reply:
sure, I added a function that lets us safely write to those, without wiping anything. I tested it as well.